xcodeproj/internal/output_files.bzl

```diff
@@ -101,7 +106,7 @@ def _create(
     # products. Until then we need them, to allow `Copy Bazel Outputs` to be
     # able to copy the products of transitive dependencies.
     transitive_products = depset(
-        direct_products if direct_products else None,
+        direct_products + dsym_files.to_list() if direct_products else None,
```

Comment:
Don't need to do `to_list()`, you can add it to `transitive` below with `+ [dsym_files]`.